services/navigation/builtin/builtin.go

```diff
@@ -277,7 +278,7 @@ func (svc *builtIn) SetMode(ctx context.Context, mode navigation.Mode, extra map
 
 	svc.mu.Lock()
 	defer svc.mu.Unlock()
-	cancelCtx, cancelFunc := context.WithCancel(context.Background())
```

Comment:
it strikes me as strange that this how the original code was set-up . is this intentional. 
from my POV this replacement seems benign, but I want to double check.

Reply:
This is intentional and correct; you'll want to undo this change.

But let's add a comment explaining why.

The `ctx` passed in has a lifespan that only runs as long as the duration of the call. So since this ctx comes from the `SetMode` call, it would be cancelled (and end this loop) once SetMode finishes setting the mode. This isn't what we want; we want it to loop indefinitely in the background until it encounters a specific error condition. So `context.Background()` is what we want here.